SPECS-SIGNED/kernel-signed-x64/kernel-signed-x64.spec

```diff
@@ -3,63 +3,103 @@
 Summary:        Signed Linux Kernel for x86_64 systems
 Name:           kernel-signed-x64
 Version:        5.10.21.1
-Release:        2%{?dist}
+Release:        3%{?dist}
 License:        GPLv2
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Kernel
 URL:            https://github.com/microsoft/CBL-Mariner-Linux-Kernel
-# This package's "version" and "release" must reflect the unsigned version that
-# was signed.
-# An important consequence is that when making a change to this package, the
-# unsigned version/release must be increased to keep the two versions consistent.
-# Ideally though, this spec will not change much or at all, so the version will
-# just track the unsigned package's version/release.
+# This spec purpose is to take an input kernel rpm and input secure-boot-signed
+# kernel binary from the same build and generate a new "kernel" rpm with the
+# signed kernel binary + all of the other original kernel files, triggers,
+# scriptlets, requires, provides, etc.
 #
-# To populate these sources:
+# We need to ensure the kernel modules and kernel binary used are from the exact
```

Comment:
I'm not sure about the idea... But you can pass different RPM macros to the build system. Why not change kernel's releases to 
`Version: %{kernelversion}%{?dist}`
provide some default version for development purposes, but for pipeline/production builds use the build number for kernelversion?

This would uniquely define the version. The versioning follows a lexicographical order so the only way for the versions to clash would be if somebody pulled a same RPM built in the same day (but even then one might override the default behavior). 

We wouldn't need to update the releases in all those specs... though now that I think of it, I think we just might utilize {?dist} instead. I think one of the reasons for its existence is situations like this.

Reply:
Also see https://docs.fedoraproject.org/en-US/packaging-guidelines/DistTag/

Reply:
I'm not sure if different RPM macros addresses the underlying problem this solution solves. The main problem concerns packages like `kernel-devel` which set its dependencies during the first kernel.spec build (`Requires: kernel = %{version}-%{release}`). This new kernel subpackage attempts to replace the original kernel RPM to fulfill the RPM dependency.

I could see RPM macros being useful in other areas like for managing the entangled specs problem.

Reply:
Wouldn't `Provides: kernel = %{version}-%{release}` solve this problem?